### PR TITLE
Refactor sync functions

### PR DESCRIPTION
**Pull request by mrogers950**
Simplify and reduce the duplication in the controller sync functions.

### CODE REVIEW

File: pkg/operator/sync_common.go [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+		return "rolebinding", false
+	}
+
+	requiredSA := resourceread.ReadServiceAccountV1OrDie(v4_00_assets.MustAsset(resourcePath + "sa.yaml"))
+	_, saModified, err := resourceapply.ApplyServiceAccount(c.corev1Client, c.eventRecorder, requiredSA)
+	if err != nil {
+		return "serviceaccount", false
+	}
+
+	return "", saModified
```
[the remark this answers is not in the document]
Comment (mrogers950) [reply pkg/operator/sync_common.go]: This was the original behavior.. perhaps we should on the others.

File: pkg/operator/sync.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	scsv1 "github.com/openshift/service-ca-operator/pkg/apis/serviceca/v1"
+)
+
+func syncControllers(c serviceCAOperator, operatorConfig *scsv1.ServiceCA) error {
+	// Sync the controller NS and the other resources. These should be mostly static.
+	err, modified := manageControllerNS(c)
+	if err != nil {
+		return fmt.Errorf("Error syncing NS: %v", err)
+	}
+	reason, modified := manageSignerControllerResources(c)
```
[the remark this answers is not in the document]
Comment (mrogers950) [reply pkg/operator/sync.go]: Yep, we want any of these changes to force all controllers to redeploy

File: pkg/operator/sync_common.go [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	scsv1 "github.com/openshift/service-ca-operator/pkg/apis/serviceca/v1"
+	"github.com/openshift/service-ca-operator/pkg/operator/operatorclient"
+	"github.com/openshift/service-ca-operator/pkg/operator/v4_00_assets"
+)
+
+func manageControllerNS(c serviceCAOperator) (error, bool) {
```
[the remark this answers is not in the document]
Comment (mrogers950) [reply pkg/operator/sync_common.go]: fixed

File: pkg/operator/sync_common.go [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+	configMap := resourceread.ReadConfigMapV1OrDie(v4_00_assets.MustAsset("v4.0.0/configmap-cabundle-controller/cm.yaml"))
+	defaultConfig := v4_00_assets.MustAsset("v4.0.0/configmap-cabundle-controller/defaultconfig.yaml")
+	requiredConfigMap, _, err := resourcemerge.MergeConfigMap(configMap, "controller-config.yaml", nil, defaultConfig, operatorConfig.Spec.ConfigMapCABundleInjectorConfig.Raw)
+	if err != nil {
+		return nil, false, err
+	}
+	return resourceapply.ApplyConfigMap(client, eventRecorder, requiredConfigMap)
+}
+
+func manageSignerControllerDeployment(client appsclientv1.AppsV1Interface, eventRecorder events.Recorder, options *scsv1.ServiceCA, forceDeployment bool) (*appsv1.Deployment, bool, error) {
```
[the remark this answers is not in the document]
Comment (mrogers950) [reply pkg/operator/sync_common.go]: I refactored those too.

**Comment by mrogers950** [comment]
@stlaz @sallyom I've rebased this on top of the recent changes.

File: pkg/operator/sync_common.go [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+func manageSignerCABundle(client coreclientv1.CoreV1Interface, eventRecorder events.Recorder) (*corev1.ConfigMap, bool, error) {
+	configMap := resourceread.ReadConfigMapV1OrDie(v4_00_assets.MustAsset("v4.0.0/apiservice-cabundle-controller/signing-cabundle.yaml"))
+	existing, err := client.ConfigMaps(configMap.Namespace).Get(configMap.Name, metav1.GetOptions{})
+	if !apierrors.IsNotFound(err) {
+		return existing, false, err
+	}
+
+	secret := resourceread.ReadSecretV1OrDie(v4_00_assets.MustAsset("v4.0.0/service-serving-cert-signer-controller/signing-secret.yaml"))
+	currentSigningKeySecret, err := client.Secrets(secret.Namespace).Get(secret.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
```
Comment (stlaz) [inline pkg/operator/sync_common.go]:
```go
if error != nil || len(currentSigningKeySecret.Data["tls.crt"]) == 0
```
handles all these three conditions

**Pull request merged**